### PR TITLE
Configuration Controller and Service Refactoring

### PR DESCRIPTION
**Pull request by wmarques**

### CODE REVIEW

File: generators/client-2/templates/src/main/webapp/app/admin/configuration/configuration.html [hunk 1]
```diff
@@ -1,41 +1,41 @@
-<div>
-    <h2 translate="configuration.title">Configuration</h2>
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/admin/configuration/configuration.html]: What about translation?

File: generators/client-2/templates/src/main/webapp/app/admin/configuration/configuration.html [hunk 2]
```diff
@@ -1,41 +1,41 @@
...
-    <span translate="configuration.filter">Filter (by prefix)</span> <input type="text" ng-model="filter" class="form-control">
-    <label ng-hide="filteredConfig.length == 0">Spring configuration</label>
+    <span>Filter (by prefix)</span> <input type="text" [(ngModel)]="filter" class="form-control">
+    <label>Spring configuration</label>
     <table class="table table-condensed table-striped table-bordered table-responsive"
-        style="table-layout:fixed" ng-hide="filteredConfig.length == 0">
+        style="table-layout:fixed">
         <thead>
         <tr>
-            <th ng-click="predicate = 'prefix'; reverse=!reverse" class="col-sm-4"><span translate="configuration.table.prefix">Prefix</span></th>
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/admin/configuration/configuration.html]: Why is sorting removed?

Comment (wmarques) [inline generators/client-2/templates/src/main/webapp/app/admin/configuration/configuration.html]: Oops I forgot that ! I'm fixing this, I'll do the same as I did in Logs page.

**Comment by sendilkumarn** [comment]
can we convert the sorting to a custom-pipe ?

**Comment by wmarques** [comment]
How can we do this ?

**Comment by sendilkumarn** [comment]
Check this http://plnkr.co/edit/FoeQ6gaP2RCjYprJXMb6?p=preview

**Comment by wmarques** [comment]
I don't know how to use a pipe with 2 params (the predicate and the reverse)..

**Comment by sendilkumarn** [comment]
@wmarques check  for 
definition --> `app/exponential-strength.pipe.ts` 
implementation --> `app/power-boost-calculator.component.ts`

**Comment by deepu105** [comment]
@wmarques is this ready for merge?

**Comment by wmarques** [comment]
@deepu105 I don't have much time right now to use a custom Pipe so I would say yes because it's functional.

**Comment by deepu105** [comment]
ok im merging it we can do the pipe later. @sendilkumarn give it a try if you hv time later

**Pull request merged**